src/main/java/org/simplejavamail/outlookmessageparser/model/OutlookRecipient.java

```diff
@@ -78,12 +81,14 @@ private void handleNameProperty(final String probablyNamePossiblyAddress) {
 	}
 
 	private void handleAddressProperty(final String probablyNamePossiblyAddress) {
-		if ((address == null || nameWasUsedAsAddress) && probablyNamePossiblyAddress.contains("@")) {
+		if (probablyNamePossiblyAddress.contains("@") && (address == null || nameWasUsedAsAddress || address.matches(X500_ADDRESS_PATTERN))) {
```

Comment:
Here "nameWasUsedAsAddress" is used and it may have not been initialised if an address property comes before a name property.
Maybe keep the default initialisation to "false" ?

Reply:
> Maybe keep the default initialisation to "false" ?

I'm not sure I understand what you mean. `nameWasUsedAsAddress` is _false_ by default.